tools/stats/upload_stats_lib.py

```diff
@@ -245,11 +245,20 @@ def unzip(p: Path) -> None:
         zip.extractall(unzipped_dir)
 
 
-def is_rerun_disabled_tests(tests: dict[str, dict[str, int]]) -> bool:
+def is_rerun_disabled_tests(
+    report: Path,
+    workflow_run_id: int,
+    workflow_run_attempt: int,
+    tests: dict[str, dict[str, int]],
+) -> bool:
     """
     Check if the test report is coming from rerun_disabled_tests workflow where
     each test is run multiple times
     """
+    job_id = get_job_id(report)
+    job_name = get_job_name(job_id, workflow_run_id, workflow_run_attempt)
+    if job_name and "rerun_disabled_tests" in job_name:
```

Comment:
Nit: I think we could do this check after the one on line 262.  This probably saves you some queries for obvious `rerun_disabled_tests` cases